### PR TITLE
Add initial documentation for TFLite frontend

### PR DESCRIPTION
**Pull request by not-jenni**
Closes https://github.com/google/iree/issues/5455

### CODE REVIEW

File: docs/website/docs/ml-frameworks/tensorflow-lite.md [hunk 1]
```diff
@@ -1,14 +1,61 @@
 # TensorFlow Lite Integration
 
-!!! todo
-    [Issue#5455](https://github.com/google/iree/issues/5455): write this documentation
+IREE supports compiling and running pre-trained TFLite models.  It converts a model to [TOSA MLIR](https://mlir.llvm.org/docs/Dialects/TOSA/), then compiles it into a VM module.
+
+## Prerequisites
+
+Download a pre-trained TFLite model from the list of [hosted models](https://www.tensorflow.org/lite/guide/hosted_models).
```
Comment (ScottTodd) [inline docs/website/docs/ml-frameworks/tensorflow-lite.md:7]: mention the TensorFlow Lite converter
also a few nits: wrap lines at 80 characters and spell out "TensorFlow Lite"

```suggestion
Download a pre-trained TensorFlow Lite model from the list of
[hosted models](https://www.tensorflow.org/lite/guide/hosted_models)
or convert your own model to a TensorFlow Lite model using the
[TensorFlow Lite converter](https://www.tensorflow.org/lite/convert).
```

File: docs/website/docs/ml-frameworks/tensorflow-lite.md [hunk 2]
```diff
@@ -1,14 +1,61 @@
 # TensorFlow Lite Integration
 
-!!! todo
-    [Issue#5455](https://github.com/google/iree/issues/5455): write this documentation
+IREE supports compiling and running pre-trained TFLite models.  It converts a model to [TOSA MLIR](https://mlir.llvm.org/docs/Dialects/TOSA/), then compiles it into a VM module.
```
Comment (ScottTodd) [inline docs/website/docs/ml-frameworks/tensorflow-lite.md:3]: I think we should spell out "TensorFlow Lite" throughout, or at least explain the abbreviation (`pre-trained TensorFlow Lite (TFLite) models`).

```suggestion
IREE supports compiling and running pre-trained TensorFlow Lite models. 
It converts a model to [TOSA MLIR](https://mlir.llvm.org/docs/Dialects/TOSA/),
then compiles it into a VM module.
```

File: docs/website/docs/ml-frameworks/tensorflow-lite.md [hunk 3]
```diff
@@ -1,14 +1,61 @@
...
+## Samples
+
+| Colab notebooks |  |
+| -- | -- |
+Text classification with TFLite and IREE | [![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/iree/blob/main/colab/tflite_text_classification.ipynb)
+
+!!! todo
 
-<!-- TODO(??): using the tflite bindings -->
+    [Issue#3954](https://github.com/google/iree/issues/3954): Add documentation for an Android demo using the [TFLite bindings](https://google.github.io/iree/bindings/tensorflow-lite/), once it is complete at [not-jenni/iree-android-tflite-demo](https://github.com/not-jenni/iree-android-tflite-demo).
```
Comment (ScottTodd) [inline docs/website/docs/ml-frameworks/tensorflow-lite.md:61]: We should introduce both the TensorFlow Lite C API and Java bindings.

File: docs/website/docs/ml-frameworks/tensorflow-lite.md [hunk 4]
```diff
@@ -1,14 +1,61 @@
...
+
+#### Using the python API
+
+``` python
+from iree.compiler import compile_str
+with open('sample.mlir') as sample_tosa_mlir:
+  compiled_flatbuffer = compile_str(sample_tosa_mlir.read(), target_backends=["vmvx"])
+```
+
+The flatbuffer can then be loaded to a VM module and run through IREE's runtime.
```
Comment (ScottTodd) [inline docs/website/docs/ml-frameworks/tensorflow-lite.md:51]: There is also a choice of bindings to use here (native or tflite)

```cmake
    "-iree-native-bindings-support=false"
    "-iree-tflite-bindings-support"
```

We'll want to link to https://google.github.io/iree/bindings/tensorflow-lite/ (when that has content), or maybe just fold that page into this one.

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/docs/ml-frameworks/tensorflow-lite.md (hunk 3 above)
Comment (not-jenni) [reply docs/website/docs/ml-frameworks/tensorflow-lite.md]: I updated the comment to mention the Java bindings.  Is there a good sample of the TensorFlow Lite C API that I could link to?

Comment (ScottTodd) [reply docs/website/docs/ml-frameworks/tensorflow-lite.md]: https://github.com/google/iree/blob/main/bindings/tflite/smoke_test.cc is all we have so far

**Review by KoolJBlack — APPROVED** [review]
Looks good. My only comment is the `!!! todo` blocks which seems unusual, but looks like they were there before. Unless scott thinks you should change them, they're fine.

**Comment by ScottTodd** [comment]
> Looks good. My only comment is the `!!! todo` blocks which seems unusual, but looks like they were there before. Unless scott thinks you should change them, they're fine.

they are fine: https://squidfunk.github.io/mkdocs-material/reference/admonitions/

File: docs/website/docs/ml-frameworks/tensorflow-lite.md [hunk 5]
```diff
@@ -1,14 +1,85 @@
 # TensorFlow Lite Integration
 
+IREE supports compiling and running pre-trained TensorFlow Lite (TFLite)
+models.  It converts a model to [TOSA MLIR](https://mlir.llvm.org/docs/Dialects/
+TOSA/), then compiles it into a VM module.
```
Comment (ScottTodd) [inline docs/website/docs/ml-frameworks/tensorflow-lite.md:5]: Do these links work as written?

To break markdown lines at 80 characters, put any `[title](some-really-long-url-that-makes-this-go-beyond-80-characters)` links on a line by themselves

```suggestion
models.  It converts a model to
[TOSA MLIR](https://mlir.llvm.org/docs/Dialects/TOSA/),
then compiles it into a VM module.
```

**Review by ScottTodd — CHANGES_REQUESTED** [review]

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**